### PR TITLE
Remove fake_eth_name_check endpoint and change fake_eth_name_check_fi…

### PR DESCRIPTION
**Pull request by djstrong**
…elds into fake_eth_name_check

### CODE REVIEW

File: api/tests/test_api.py [hunk 1]
```diff
@@ -882,135 +882,6 @@ def test_primary_name_get_emoji(test_client):
     )
 
 
-@pytest.mark.flaky(retries=2, condition=not pytest.use_monkeypatch)
```
[the remark this answers is not in the document]
Comment (djstrong) [reply api/tests/test_api.py:885]: @lightwalker-eth I have added link to these tests in the story.

File: apps/website/src/app/page.tsx [hunk 2]
```diff
@@ -54,8 +54,9 @@ const fakeEthNameCheck = `import { nameguard } from "@namehash/nameguard";
 const contract = "0x8Ae0e6dd8eACe27045d9e017C8Cf6dAa9D08C776";
 const tokenId = "619959211285214429591066501314626337448852696
                  24153038309795231243542768648193";
+const fields = {"title": "nick.eth"}
```
Comment (notrab) [inline apps/website/src/app/page.tsx:57]:
```suggestion
```

File: docs/api-reference/fake-eth-name-check.mdx [hunk 3]
```diff
@@ -1,19 +1,27 @@
...
+  "contract_address": "0x495f947276749ce646f68ac8c248420045cb7b5e"
+  "token_id": "61995921128521442959106650131462633744885269624153038309795231243542768648193"
+  "fields": {"title": "nick.eth"}
+}'
 ```
 
 ```ts SDK
 const contractAddress = "";
 const tokenId = "";
+const fields = "";
```
Comment (notrab) [inline docs/api-reference/fake-eth-name-check.mdx:22]:
```suggestion
```

File: docs/api-reference/fake-eth-name-check.mdx [hunk 4]
```diff
@@ -1,19 +1,27 @@
...
 
-const report = await nameguard.fakeEthNameCheck(contractAddress, tokenId);
+const report = await nameguard.fakeEthNameCheck(contractAddress, tokenId, fields);
```
Comment (notrab) [inline docs/api-reference/fake-eth-name-check.mdx:24]:
```suggestion
const report = await nameguard.fakeEthNameCheck(contractAddress, tokenId, {
  title: "..."
});
```

File: apps/website/src/app/page.tsx [hunk 5]
```diff
@@ -55,7 +55,9 @@ const contract = "0x8Ae0e6dd8eACe27045d9e017C8Cf6dAa9D08C776";
 const tokenId = "619959211285214429591066501314626337448852696
                  24153038309795231243542768648193";
 
-const result = await nameguard.fakeEthNameCheck(contract, tokenId);`;
+const result = await nameguard.fakeEthNameCheck(contract, tokenId, {
+  title: "nick.eth"
+});`;
```
Comment (notrab) [inline apps/website/src/app/page.tsx:60]:
```suggestion
});
```

File: packages/sdk/src/index.ts [hunk 6]
```diff
@@ -672,12 +672,14 @@ class NameGuard {
...
    * @param {string} contract_address Contract address for the NFT contract (ERC721 and ERC1155 supported).
    * @param {string} token_id The ID of the token (in hex or decimal format).
+   * @param {string} fields Fields with values which will be investigated (e.g. title, collection name, metadata) whether they look like fake .eth ENS name. `title` key is mandatory, for ENS contracts it should be the ENS name.
    * @param {FakeEthNameOptions} options The options for the fake .eth ens name check.
    * @returns {Promise<FakeEthNameCheckResult>}  A promise that resolves with a `FakeEthNameCheckResult`.
    */
   public fakeEthNameCheck(
     contract_address: string,
     token_id: string,
+    fields: object,
```
Comment (notrab) [inline packages/sdk/src/index.ts:682]:
```suggestion
    fields: Record<string, string>,
```

File: packages/sdk/src/index.ts [hunk 7]
```diff
@@ -672,12 +672,14 @@ class NameGuard {
...
    * @param {string} contract_address Contract address for the NFT contract (ERC721 and ERC1155 supported).
    * @param {string} token_id The ID of the token (in hex or decimal format).
+   * @param {string} fields Fields with values which will be investigated (e.g. title, collection name, metadata) whether they look like fake .eth ENS name. `title` key is mandatory, for ENS contracts it should be the ENS name.
    * @param {FakeEthNameOptions} options The options for the fake .eth ens name check.
    * @returns {Promise<FakeEthNameCheckResult>}  A promise that resolves with a `FakeEthNameCheckResult`.
    */
   public fakeEthNameCheck(
     contract_address: string,
     token_id: string,
+    fields: Record<string, string>,
```
Comment (notrab) [inline packages/sdk/src/index.ts:682]: We should probably check that `fields` was present with at least one key.

```ts
if (Object.keys(fields).length === 0) {
  throw new Error("The 'fields' object must contain at least one field.");
}
```

Comment (djstrong) [reply packages/sdk/src/index.ts:682]: It must contain `title` field.

Comment (notrab) [reply packages/sdk/src/index.ts:682]: Good spot! Let's do two things:

1. Update `Record<string, string>` to be:

```ts
interface FieldsWithRequiredTitle extends Record<string, string> {
    title: string;
}

// ...

fields: FieldsWithRequiredTitle
```

2. Update the validation to be:

```ts
if (!fields.title || typeof fields.title !== 'string') {
  throw new Error("The 'fields' object must contain a 'title' key with a string value.");
}
```

File: packages/sdk/src/index.ts [hunk 8]
```diff
@@ -131,7 +131,7 @@ export interface FakeEthNameCheckResult {
    *
    * `null` if `status` is `unknown_nft`
    */
-  investigated_fields: object | null; // TODO: dict[str,str] | null
+  investigated_fields: Record<string, string> | null;
```
Comment (notrab) [inline packages/sdk/src/index.ts:134]: Could we make use of the same interface below? `FieldsWithRequiredTitle`?

Comment (notrab) [reply packages/sdk/src/index.ts:134]: Actually no, that's for input. My bad!

**Review by notrab — APPROVED** [review]

**Pull request merged**